draft-ietf-rats-corim.md

```diff
@@ -1207,6 +1207,242 @@ The following describes each member of the `concise-bom-tag` map.
+
+All available CoRIMs are collected.
+A Verifier may be pre-configured with a large number of tags describing many
+types of device.
+All CoRIMs are loaded at this stage, later stages will select the CoRIMs
+appropriate to the Evidence Appraisal step.
+
+CoRIMs that are not within their validity period, or that cannot be associated
+with an authenticated and authorised source MUST be discarded.
+
+CoRIM that are secured by a cryptographic mechanism such as a signature which
+does not pass validation MUST be discarded.
+
+Other selection criteria MAY be applied.
+
+For example, if the Evidence format is known in advance, tags that do not match
+the expected profile can be readily discarded.
+
+The selection process MUST yield at least one usable tag.
+
+### CoBOM Extraction
+
+All the available Concise Bill Of Material (CoBOMs) tags are then collected
+from the selected CoRIMs.
+
+The Verifier MUST activate all tags referenced by a CoBOM.
+
+After the Verifier has processed all CoBOMs it MUST discard any tags which have
+not been activated by a CoBOM.
+
+### Tags Identification and Validation
+
+The Verifier chooses tags -- including Concise Module ID Tags (CoMID, {{sec-comid}}),
+Concise Software ID Tags (CoSWID, {{-coswid}}),
+and/or Concise Trust Anchor Stores (CoTS, {{?I-D.ietf-rats-concise-ta-stores}}) --
+from the selected CoRIMs.
+
+The Verifier MUST discard all tags which are not syntactically and semantically
+valid.
+In particular, any cross-referenced triples (e.g., CoMID-CoSWID linking triples)
+MUST be successfully resolved.
+
+### Appraisal Context Construction
+
+All of the validated and potentially useful tags are loaded into the Appraisal Context.
+Each tag is loaded into the Appraisal Context as a single unit, later stages
+will accept or ignore the whole tag.
+
+This concludes the initialisation phase.
+
+## Evidence Collection
+
+In the evidence collection phase the Verifier communicates with attesters to
+collect evidence.
+
+The first part of the Evidence collection phase does not perform any
+cryptographic validation.
+This allows Verifiers to use untrusted code for their initial Evidence collection.
+
+The results of the evidence collection are protocol specific data and transcripts
+which are used as input to appraisal by the Verifier.
+
+### Cryptographic validation of Evidence
+
+If the authenticity of Evidence is secured by a cryptographic mechanism such as
+a signature, the first step in the Evidence Appraisal is to perform
+cryptographic validation of the Evidence.
+
+The exact cryptographic signature validation mechanics depend on the specific
+Evidence collection protocol.
+
+For example:
+In DICE, a proof of liveness is performed on the final key in the certificate
+chain.
+If this passes then a suitable certification path anchored on a trusted root
+certificate is looked up -- e.g., based on linking information obtained from
+the DeviceID certificate (see Section 9.2.1 of {{DICE-Layering-Architecture}}) --
```

Comment:
remember to add the bib entry for DICE-Layering-Architecture or it won't compile.

Reply:
DICE Layering Architecture, Version 1.0, Revision 0.19, July 23, 2020. https://trustedcomputinggroup.org/wp-content/uploads/DICE-Layering-Architecture-r19_pub.pdf